### PR TITLE
Fix new rnn() mask keyword argument in softmax()

### PR DESCRIPTION
**Pull request by sgerke**
In `activations.py`, the changes introduced in commit ac773ed243154c5408937122bb67f5ec754b29d5 (renaming "mask" to "masking" were not aligned in `keras/activations.py:softmax()` ), causing an error when using softmax activation.

### CODE REVIEW

**Comment by fchollet** [comment]
You are right, but that does not fix the issue: another problem was the new RNN API, with the `output_dim` argument. I fixed it and added a test.
